src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-bucket-tiering-form/rgw-bucket-tiering-form.component.ts

```diff
@@ -56,11 +57,31 @@ export class RgwBucketTieringFormComponent extends CdForm implements OnInit {
   ngOnInit() {
     this.rgwBucketService
       .getLifecycle(this.bucket.bucket, this.bucket.owner)
-      .subscribe((lifecycle) => {
-        this.configuredLifecycle = lifecycle || { LifecycleConfiguration: { Rules: [] } };
+      .subscribe((lifecycle: Lifecycle) => {
+        if (lifecycle === null) {
+          lifecycle = { LifecycleConfiguration: { Rule: [] } };
+        }
+        if (!Array.isArray(lifecycle.LifecycleConfiguration.Rule)) {
+          lifecycle.LifecycleConfiguration.Rule = [lifecycle.LifecycleConfiguration.Rule];
```

Comment:
Ideally we should always get array either empty or with data from backend. Cant this be changed from api ?

Reply:
I am seeing so many isArray checks otherwise as well ?